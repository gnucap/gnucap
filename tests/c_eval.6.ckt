* parameter and string eval check
.eval "hello"
.eval "hello" + " " + "world"

.parameter string what="world";
.eval "hello " + what
.eval "hello " + (what + 1 + 2) + 3 + 4

.parameter a = 1
.eval a
.parameter b = a;
.eval b

.parameter string a = "b";
.eval a
.parameter string b = a;
.eval b
.parameter string c = {b + "."};
.eval c

* does not work, use verilog
.parameter string d = b + ".";
.eval d

.verilog
module test();
	parameter string d = b + "."; // works.
endmodule

.end
