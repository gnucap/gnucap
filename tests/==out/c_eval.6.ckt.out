* parameter and string eval check 
"hello"="hello"
(("hello" + " ") + "world")="hello world"
("hello " + what)="hello world"
((("hello " + ((what + 1) + 2)) + 3) + 4)=(("hello " + ("world" + 3)) + 7)
a= 1.
b= 1.
a="b"
b="b"
c="b."
.parameter string d = b + "."; 
                        ^ ? syntax error
d="b"
